platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/UnsignedEvent.java

```diff
@@ -372,171 +96,21 @@ public String toString() {
-        return !hasSelfParent()
-                ? otherParents
-                : Stream.concat(Stream.of(selfParent), otherParents.stream()).toList();
-    }
-
-    /**
-     * Get the self parent generation
-     *
-     * @return the self parent generation
-     */
-    public long getSelfParentGen() {
-        if (selfParent == null) {
-            return EventConstants.GENERATION_UNDEFINED;
-        }
-        return selfParent.eventDescriptor().generation();
-    }
-
-    /**
-     * Get the maximum generation of the other parents.
-     *
-     * @return the maximum generation of the other parents
-     * @deprecated this method should be replaced since there can be multiple other parents.
-     */
-    @Deprecated
-    public long getOtherParentGen() {
-        if (otherParents == null || otherParents.isEmpty()) {
-            return EventConstants.GENERATION_UNDEFINED;
-        }
-        if (otherParents.size() == 1) {
-            return otherParents.getFirst().eventDescriptor().generation();
-        }
-        // 0.46.0 adds support for multiple other parents in the serialization scheme, but not yet in the
-        // implementation. This exception should never be reached unless we have multiple parents and need to
-        // update the implementation.
-        throw new UnsupportedOperationException("Multiple other parents is not supported yet");
-    }
-
-    /**
-     * Get the hash of the self parent.
-     *
-     * @return the hash of the self parent
-     */
-    @Nullable
-    public Hash getSelfParentHash() {
-        if (selfParent == null) {
-            return null;
-        }
-        return selfParent.hash();
-    }
-
-    /**
-     * Get the hash of the other parent with the maximum generation.
-     *
-     * @return the hash of the other parent with the maximum generation
-     */
-    @Nullable
-    @Deprecated
-    public Hash getOtherParentHash() {
-        if (otherParents == null || otherParents.isEmpty()) {
-            return null;
-        }
-        if (otherParents.size() == 1) {
-            return otherParents.getFirst().hash();
-        }
-        // 0.46.0 adds support for multiple other parents in the serialization scheme, but not yet in the
-        // implementation. This exception should never be reached unless we have multiple parents and need to
-        // update the implementation.
-        throw new UnsupportedOperationException("Multiple other parents is not supported yet");
-    }
-
-    /**
-     * Check if the event has a self parent.
-     *
-     * @return true if the event has a self parent
-     */
-    public boolean hasSelfParent() {
-        return selfParent != null;
-    }
-
-    /**
-     * Check if the event has other parents.
-     *
-     * @return true if the event has other parents
-     */
-    public boolean hasOtherParent() {
-        return otherParents != null && !otherParents.isEmpty();
+    public SoftwareVersion getSoftwareVersion() { // TODO remove this method
```

Comment:
TODO statement says to remove this method.

Reply:
removed